pkg/tkctl/cmd/check/check.go

```diff
@@ -0,0 +1,460 @@
+// Copyright 2019. PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package check
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/Masterminds/semver"
+	"github.com/fatih/color"
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap.com/v1alpha1"
+	config_v2_1_16 "github.com/pingcap/tidb-operator/pkg/config/v2_1_16"
+	"github.com/pingcap/tidb-operator/pkg/tkctl/config"
+	"github.com/pingcap/tidb-operator/pkg/util"
+	"github.com/spf13/cobra"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/helm/pkg/chartutil"
+	"k8s.io/helm/pkg/engine"
+	helm_env "k8s.io/helm/pkg/helm/environment"
+	chartproto "k8s.io/helm/pkg/proto/hapi/chart"
+	"k8s.io/helm/pkg/timeconv"
+	tversion "k8s.io/helm/pkg/version"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	kubeVersion   = "v1.12.8"
+	checkLongDesc = `
+                  Preflight check for TiDB cluster configuration.
+`
+	checkExample = `
+      # check config from local chart
+      tkctl check -t demo --chart charts/tidb-cluster --values charts/tidb-cluster/values.yaml
+
+      # check config from remote chart
+      tkctl check -t demo --chart pingcap/tidb-cluster --values charts/tidb-cluster/values.yaml --version v1.1.0-alpha.2
+`
+	// AWS load balancer annotations
+	awsILBAnnotationKey    = "service.beta.kubernetes.io/aws-load-balancer-internal"
+	awsILBAnnotationVal    = "0.0.0.0/0"
```

Comment:
The value for internal LB is "true" on [k8s doc](https://kubernetes.io/docs/concepts/services-networking/#loadbalancer)
[...]
metadata:
    name: my-service
    annotations:
        service.beta.kubernetes.io/aws-load-balancer-internal: "true"
[...]

Reply:
From AWS [document](https://docs.aws.amazon.com/eks/latest/userguide/load-balancing.html), its value is `0.0.0.0/0`. And we've already tested and used this configuration in benchmark environment.